_ext/eclipse-base/src/main/java/com/diffplug/spotless/extra/eclipse/base/SpotlessEclipseFramework.java

```diff
@@ -0,0 +1,227 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.extra.eclipse.base;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.xml.parsers.SAXParserFactory;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleException;
+
+import com.diffplug.spotless.extra.eclipse.base.osgi.BundleController;
+import com.diffplug.spotless.extra.eclipse.base.runtime.PluginRegistrar;
+
+/** Setup a framework for Spotless Eclipse based formatters */
+public final class SpotlessEclipseFramework {
+	/** Spotless always uses \n internally as line delimiter */
+	public static final String LINE_DELIMITER = "\n";
+
+	/**
+	 * Default internal bundles required by most plugins.
+	 * <p>
+	 * Services provided by an internal bundle are not accessed via the plugin registry, but by static methods.
+	 * </p>
+	 */
+	public enum DefaultBundles {
+		/** Plugins ask the platform whether core runtime bundle is in debug mode. Note that the bundle requires the EnvironmentInfo service. */
+		PLATFORM(org.eclipse.core.internal.runtime.PlatformActivator.class),
+		/** The Spotless BundleController wraps the OSGi layer. But the plugin/extension look-up still uses the Eclipse OSGi registry.*/
+		REGISTRY(org.eclipse.core.internal.registry.osgi.Activator.class),
+		/** Eclipse preferences always check whether this bundle has been activated before preference are set.*/
+		PREFERENCES(org.eclipse.core.internal.preferences.Activator.class),
+		/** The common runtime provides provides common services, like log and service adapters registry. */
+		COMMON(org.eclipse.core.internal.runtime.Activator.class);
+
+		private final Class<? extends BundleActivator> activatorClass;
+
+		private DefaultBundles(Class<? extends BundleActivator> clazz) {
+			activatorClass = clazz;
+		}
+
+		/** Create new bundle activator instance. */
+		public BundleActivator create() {
+			return createInstance(activatorClass);
+		}
+
+		/** Create bundle activator instances for all enumerated values. */
+		public static Collection<BundleActivator> createAll() {
+			return Arrays.stream(values()).map(value -> value.create()).collect(Collectors.toList());
+		}
+	}
+
+	/**
+	 * Default plugins required by most Spotless formatters.
+	 * <p>
+	 * Eclipse plugins are OSGI bundles itself and do not necessarily derive from the Eclipse Plugin class.
+	 * BundleActivator implementation may as well server as plugins.
```

Comment:
s/BundleActivator/{@code BundleActivator}/ ?

Reply:
s/server/serve/ ?

Reply:
:-1: I do not agree to the `{@code ...` usage.
In my opinion the JavaDoc does not profit a lot by the usage and I always find it distracting and too verbose when I study code. The IDE anyway supports the linkage to the class sources with or without it.

Reply:
> :-1: I do not agree to the `{@code ...` usage.
> In my opinion the JavaDoc does not profit a lot by the usage and I always find it distracting and too verbose when I study code. The IDE anyway supports the linkage to the class sources with or without it.

Ah okay. I'm personally a fan of it, but I don't want to push things if @nedtwigg isn't a fan of it either.

@nedtwigg Do you have a preference regarding this?